multihost/db-postgresql-with-custom-certificates/test.sh

```diff
@@ -0,0 +1,375 @@
+        rlRun "cp $(x509Cert tenant) $CERTDIR/tenant-cert.pem"
+        rlRun "cp $(x509Key tenant) $CERTDIR/tenant-key.pem"
+        # assign cert ownership to keylime user if it exists
+        id keylime && rlRun "chown -R keylime:keylime $CERTDIR"
+        rlRun "mkdir -p /var/lib/keylime/cv_ca/"
+        rlRun "cp $CERTDIR/cacert.pem /var/lib/keylime/cv_ca/"
+
+        rlServiceStop postgresql
+        rlRun "cat > setup.psql <<_EOF
+create database verifierdb;
+create user verifier with encrypted password 'fire';
+grant all privileges on database verifierdb to verifier;
+create database registrardb;
+create user registrar with encrypted password 'regi';
+grant all privileges on database registrardb to registrar;
+\connect verifierdb;
+grant all privileges on schema public to verifier;
+\connect registrardb;
+grant all privileges on schema public to registrar;
+_EOF"
+
+        rlRun "chmod a+x setup.psql"
+        #need set up permission for running postgres
+        rlRun "chmod a+x /tmp/tmp.*"
+        rlFileBackup --clean --missing-ok /var/lib/pgsql /etc/postgresql-setup
+        rlRun "rm -rf /var/lib/pgsql/data"
+        rlRun "postgresql-setup --initdb --unit postgresql"
+        # configure user authentication with md5 and pgsql listening address
+        rlRun "sed -i \"s|^host.*all.*all.*127.0.0.1.*ident|host all all 0.0.0.0/0 md5|\" /var/lib/pgsql/data/pg_hba.conf"
+        rlRun "sed -i \"s|^#listen_addresses.*|listen_addresses = '0.0.0.0'|\" /var/lib/pgsql/data/postgresql.conf"
+        rlServiceStart postgresql
+        sleep 3
+        rlRun "sudo -u postgres psql -f setup.psql"
+        rlRun "sync-set ATTESTATION_SERVER_GENERATE_CERTS_DONE"
+
+        # tenant config
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        rlRun "limeUpdateConf tenant verifier_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant registrar_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant tls_dir $CERTDIR"
+        rlRun "limeUpdateConf tenant trusted_server_ca '[\"cacert.pem\"]'"
+        rlRun "limeUpdateConf tenant client_cert tenant-cert.pem"
+        rlRun "limeUpdateConf tenant client_key tenant-key.pem"
+
+        #copy cert to agent machine
+        rlRun "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null $CERTDIR/cacert.pem root@$AGENT_IP:/var/lib/keylime"
+        rlRun "sync-block ANSIBLE_SETUP_DONE ${SETUP_SERVER_IP}" 0 "Waiting for ansible setup of system roles."
+        sleep 5
+        rlRun "limeWaitForVerifier"
+        rlRun "limeWaitForRegistrar"
+
+        #confirm that verifier and registrar accessed and modified databases
+        rlRun -s "sudo -u postgres psql -c 'SELECT datname FROM pg_database;'"
+        rlAssertGrep "verifierdb" $rlRun_LOG
+        rlAssertGrep "registrardb" $rlRun_LOG
+
+        rlRun "sync-set ATTESTATION_SERVER_START"
+        rlRun "sync-block AGENT_STARTS ${AGENT_IP}" 0 "Waiting for ansible setup of system roles."
```

Comment:
In fact waiting for keylime agent start?

Reply:
That's leftover. I'll update description.